src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetPackageSearchService.cs

```diff
@@ -212,12 +212,12 @@ public NuGetPackageSearchService(ServiceActivationOptions options, IServiceBroke
             // Transitive packages will have only one version the first time they are loaded, when the package is selected we update the cache with all the versions
             if (backgroundDataCache != null)
             {
-                if (isTransitive &&
-                    (backgroundDataCache.AllVersionsContextInfo.Result == null || backgroundDataCache.AllVersionsContextInfo.Result.Count <= 1))
+                if (isTransitive && (backgroundDataCache.AllVersionsContextInfo.Result == null || backgroundDataCache.AllVersionsContextInfo.Result.Count <= 1)
+                    || backgroundDataCache.AllVersionsContextInfo.Result.First().PackageSearchMetadata == null) // If the item was cached with search API, PackageSearchMetadata could be null. If so, update it with registration api information
```

Comment:
```suggestion
                // If the item was cached with search API, PackageSearchMetadata could be null. If so, update it with registration api information
                if (isTransitive
                    && !(backgroundDataCache.AllVersionsContextInfo.Result?.Count > 1)
                    || backgroundDataCache.AllVersionsContextInfo.Result?.First().PackageSearchMetadata == null) 
```

I don't know if this increases readability, I think it might?  Hopefully I did the boolean logic correctly.